src/storage/src/row_serde/value_serde.rs

```diff
@@ -114,9 +114,10 @@ impl ValueRowSerdeNew for ColumnAwareSerde {
                     // It's okay since we previously banned impure expressions in default columns.
                     build_from_prost(&expr.expect("expr should not be none"))
                         .expect("build_from_prost error")
-                        .eval_row_infallible(&OwnedRow::empty())
+                        .eval_row(&OwnedRow::empty())
                         .now_or_never()
                         .expect("constant expression should not be async")
+                        .expect("eval_row failed")
```

Comment:
Could you please explain a little bit on why this can be changed to fallible eval?

Reply:
It was originally fallible evaluation that panics. 🙁 In the last PR it was changed to a `eval_infallible` on top of a normal expression that is not built with `non_strict` builders, which was still potentially to panic.